lib/vdc/BOVCollection.cpp

```diff
@@ -232,6 +263,18 @@ int BOVCollection::_validateParsedValues()
         _byteOffset = _tmpByteOffset;
         _byteOffsetAssigned = true;
     }
+
+    // Validate file size
+    size_t      fileSize;
```

Comment:
This validation seems overly restrictive: it requires the file to be the exact same size as the volume. This would, for example, prevent someone from storing multiple volumes in the same file and using the 'offset' to access the different volumes (variables).

Secondly, why is this check even necessary at all If there is a short file that should be detected when the file is read. Catching this error here seems superfluous. Am I missing something?

Reply:
@shaomeng and I were discussing this in the #2791 thread.  I think this is a question for him.

Reply:
I believe errors of the following nature should be caught: when the code wants to read in `N` bytes and there are `M` bytes left in the file where `M < N`.

How to catch these errors: `std::fread()` returns how many values it successfully reads in. Compare the return value with the expected number of values to read.

Reply:
I think @shaomeng and i are  both saying that these errors can be caught by the return value from fread, and that it is not necessary to stat the file before reading it.

Reply:
Fixed at the end of the _validateParsedValues function.

Reply:
Please see comments above regarding removing file size validation entirely from this function. If there is an error reading the file it will be caught when the user actually attempts to access the file. This is consistent with what we do elsewhere.

Reply:
Fixed.